pkg/controller/controllercmd/builder.go

```diff
@@ -40,8 +41,11 @@ type ControllerContext struct {
 	// Note that this config might not be safe for CR resources, instead it should be used for other resources.
 	ProtoKubeConfig *rest.Config
 
+	// EventRecorder is used to record events in controllers.
 	EventRecorder events.Recorder
-	Context       context.Context
+
+	// Done returns a channel which will close on termination.
+	Done func() <-chan struct{}
```

Comment:
make the channel a private field and add a method?

Reply:
sure, refactored.

Reply:
as we discussed, this is not perfect and we should probably pass the context as the first arg in RunControllers, but i don't want to groom the scope for this PR.